.github/workflows/check.yml

```diff
@@ -1,53 +1,82 @@
-name: check
+name: Check
 
-on: [pull_request, push]
+on:
+  push:
+  pull_request:
+    branches:
+      - master
 
 jobs:
+  check_duplicate_runs:
+    name: Check for duplicate runs
+    continue-on-error: true
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          concurrent_skipping: always
+          cancel_others: true
+          skip_after_successful_duplicate: true
+          paths_ignore: '["**/README.md", "**/CHANGELOG.md", "**/LICENSE.md"]'
+          do_not_skip: '["pull_request"]'
+
   mix_check:
     name: mix check (Elixir ${{matrix.elixir}}, OTP ${{matrix.otp}})
+    runs-on: ubuntu-18.04
+    needs: check_duplicate_runs
+    if: ${{ needs.check_duplicate_runs.outputs.should_skip != 'true' }}
 
     strategy:
       matrix:
-        include:
-          - elixir: '1.10.4'
-            otp: '23.0.2'
-          - elixir: '1.7.0'
-            otp: '19.0'
-
-    runs-on: ubuntu-18.04
+        elixir:
```

Comment:
Adding all Elixir versions to the matrix instead of just lowest and highest seems like a step in the right direction, although it's a risky move for a quite shameful reason - since it seems that test suite currently suffers some random failures - some tests hang forever on the CI, I didn't replicate it locally.

Reply:
Well, it looks like tests are failing not only on Elixir versions between 1.7 and 1.11, but the latest 1.11 too.
It looks like there is a race condition when sample projects tries to recompile ex_check to run all the checks, but cannot do that because compiled file is already blocked by a running application.

Reply:
That sounds right, although I don't quite understand - every sample project and its build artifact lands in its own temporary dir, do you mean that ex_check source is locked? If so then we could change test cases to copy ex_check source per-project as well. 

As a last resort we could disable async testing, but I'd rather avoid that.

Reply:
I finally fixed it.

1. Instead of using the repo itself as a path for test project dependency, a separated copy is being created.
2. The issue with `:crypto.strong_rand_bytes` was caused by a fact, this function needs a source of entropy to work., like CPU thermometer or other sensor with truly random data. But it looks like there is not enough entropy in the system random pool, so the process is waiting for these random bits and then being halted by a ExUnit timeout. I've replaced it with a less random, but this is definitely enough to create few folders with a random name.

Reply:
👏